Sources/eppo/RuleEvaluator.swift

```diff
@@ -175,95 +240,131 @@ public class FlagEvaluator {
+                    conditionDouble = doubleValue
+                } else {
+                    return false
+                }
+                
+                switch condition.operator {
+                case .greaterThanEqual:
+                    return valueDouble >= conditionDouble
+                case .greaterThan:
+                    return valueDouble > conditionDouble
+                case .lessThanEqual:
+                    return valueDouble <= conditionDouble
+                case .lessThan:
+                    return valueDouble < conditionDouble
+                default:
                     return false
                 }
-            case .matches:
-                return Compare.matchesRegex(
-                    try value.toEppoString(),
-                    try condition.value.toEppoString()
-                )
-            case .notMatches:
-                return !Compare.matchesRegex(
-                    try value.toEppoString(),
-                    try condition.value.toEppoString()
-                )
-            case .oneOf:
-                return try Compare.isOneOf(
-                    value.toEppoString(),
-                    condition.value.getStringArrayValue()
-                )
-            case .notOneOf:
-                return try !Compare.isOneOf(
-                    value.toEppoString(),
-                    condition.value.getStringArrayValue()
-                )
-            default:
-                return false;
             }
-        } catch {
-            // Handle or log the error
+        case .matches, .notMatches:
+            if let conditionString = try? condition.value.toEppoString(),
+               let valueString = try? value.toEppoString() {
+                
+                if isConfigObfuscated,
+                   let decoded = base64Decode(conditionString) {
+                    return condition.operator == .matches ? Compare.matchesRegex(valueString, decoded) : !Compare.matchesRegex(valueString, decoded)
+                } else {
+                    return condition.operator == .matches ? Compare.matchesRegex(valueString, conditionString) : !Compare.matchesRegex(valueString, conditionString)
+                }
+            }
+            
             return false
+        case .oneOf, .notOneOf:
+            if let valueString = try? value.toEppoString(),
+               let conditionArray = try? condition.value.getStringArrayValue() {
+                
+                if isConfigObfuscated {
+                    let valueStringHash = getMD5Hex(valueString)
+                    return condition.operator == .oneOf ? Compare.isOneOf(valueStringHash, conditionArray) : !Compare.isOneOf(valueStringHash, conditionArray)
+                } else {
+                    return condition.operator == .oneOf ? Compare.isOneOf(valueString, conditionArray) : !Compare.isOneOf(valueString, conditionArray)
+                }
```

Comment:
Same here, we just need to figure out the string we're looking for and then the other code is the same.

Example in [Android SDK](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/main/java/cloud/eppo/android/RuleEvaluator.java#L160).